### DISCUSSION TITLE
Solution for .nvmrc

### DISCUSSION THREAD

**Discussion by ronhillel** [coreybutler/nvm-windows#994] [Ideas & Feature Requests]
We cannot do nvm use on windows without the version number but we can setup it up for our projects so we don't need to bother about it every time we run them.

My solution is creating new files:
**.nvmrc
.nvmrc-script.bash**

and adding a pre-script to the file **package.json**

e.g:
**.nvmrc file content:**
`v19.4.0`

**.nvmrc-script.bash file content:**
```
# setup a variable called NODE_VERSION
NODE_VERSION=$(echo $(cat .nvmrc))

# use the variable to switch to the correct node version (installing if necessary)
nvm install $NODE_VERSION
nvm use $NODE_VERSION
```

**new pre-script:**
`"prestart": "bash .nvmrc-script.bash",`

Enjoy.

**Comment by bronsonmock** [comment]
## Git bash shell
If you want a project independent solution, you can create an [alias function](https://github.com/coreybutler/nvm-windows/issues/16#issuecomment-1679581200) in your .bashrc file. If you don't need to get around the `mklink` issue, then you can simplify the alias to:

```shell
function nvm() {
    if [ $# -eq 1 ] && [[ "$1" == @(install|use) ]]; then
        command nvm "$1" $(cat .nvmrc)
    else
        command nvm "$@"
    fi
}
```

Then you can just call the commands in your shell after switching to the project root:
```shell
nvm install
nvm use
```

## npm script
If you still want to use an npm script, you can accomplish that without adding an additional script file:

1. Using the function alias (requires `export -f nvm` to be added to .bashrc):
    ```shell
    "prestart": "sh -c 'nvm install' && sh -c 'nvm use'"
    ```
2. No alias, pure bash:
    ```shell
    "prestart": "sh -c 'nvm install $(cat .nvmrc)' && sh -c 'nvm use $(cat .nvmrc)'"
    ```
